browse-next/src/components/DeviceSetupDefineMask.vue

```diff
@@ -0,0 +1,823 @@
+    const regionData = regionsArray.value[i].regionData;
+    const firstPoint = regionData[0];
+    const text = `${i + 1}`;
+    const textMetrics = context.measureText(text);
+    const textWidth = textMetrics.width;
+    const circleRadius = devicePixelRatio.pixelRatio.value * 14; // Radius based on half of the text width
+
+    const circleX = firstPoint.x * canvas.value.width;
+    const circleY =
+      firstPoint.y * canvas.value.height +
+      2 * devicePixelRatio.pixelRatio.value;
+
+    context.fillStyle = "rgba(13, 110, 253)";
+    context.beginPath();
+    context.arc(circleX, circleY, circleRadius, 0, Math.PI * 2);
+    context.closePath();
+    context.fill();
+
+    context.fillStyle = "#ffffff";
+    context.fillText(
+      text,
+      circleX - textMetrics.width / 2,
+      circleY + fontSize / 2
+    );
+  }
+}
+
+function drawPolygon(): void {
+  const canvasElement = canvas.value;
+  const context = canvasElement.getContext("2d");
+  context.lineWidth = 4 * devicePixelRatio.pixelRatio.value;
+  context.strokeStyle = "rgba(13, 110, 253, 1)";
+  context.beginPath();
+  for (let i = 0; i < points.value.length; i++) {
+    const x = points.value[i].x * canvas.value.width;
+    const y = points.value[i].y * canvas.value.height;
+    if (i === 0) {
+      context.moveTo(x, y);
+    } else {
+      context.lineTo(x, y);
+    }
+  }
+  context.stroke();
+
+  context.fillStyle = "rgba(13, 110, 253, 1)";
+  for (let i = 0; i < points.value.length; i++) {
+    const x = points.value[i].x * canvas.value.width;
+    const y = points.value[i].y * canvas.value.height;
+    context.beginPath();
+    context.arc(x, y, 4 * devicePixelRatio.pixelRatio.value, 0, Math.PI * 2);
+    context.closePath();
+    context.fill();
+  }
+}
+
+function isPolygonClosed(): boolean {
+  if (points.value.length < 2) {
+    return false;
+  }
+  const firstPoint = points.value[0];
+  const lastPoint = points.value[points.value.length - 1];
+  if (
+    !firstPoint ||
+    !lastPoint ||
+    !("x" in firstPoint) ||
+    !("x" in lastPoint)
+  ) {
+    return false;
+  }
+
+  const distance = Math.sqrt(
+    Math.pow(
+      firstPoint.x * canvas.value.width - lastPoint.x * canvas.value.width,
+      2
+    ) +
+      Math.pow(
+        firstPoint.y * canvas.value.height - lastPoint.y * canvas.value.height,
+        2
+      )
+  );
+
+  if (distance < polygonClosedTolerance.value) {
+    points.value[points.value.length - 1] = points.value[0];
+  }
+
+  clearMask(); //causes them to snap as desired, however recursion error
+  generateMask();
+  drawPolygon();
+  return distance < polygonClosedTolerance.value;
+}
+
+const toggleAreaSelect = () => {
+  updateExistingMaskRegions();
+  selectingArea.value = !selectingArea.value;
+  points.value = [];
+  clearMask();
+  generateMask();
+};
+
+const toggleMaskEnabled = () => {
+  maskEnabled.value = !maskEnabled.value;
+  if (maskEnabled.value) {
+    points.value = [];
+    clearMask();
```

Comment:
`clearMask` gets called in either case, so maybe move it outside the conditional?